packages/material-ui-lab/src/overrides.d.ts

```diff
@@ -0,0 +1,72 @@
+import type { AlertClassKey, AlertProps } from './Alert';
+import type { AlertTitleClassKey, AlertTitleProps } from './AlertTitle';
+import type { AutocompleteClassKey, AutocompleteProps } from './Autocomplete';
+import type { AvatarGroupClassKey, AvatarGroupProps } from './AvatarGroup';
+import type { PaginationClassKey, PaginationProps } from './Pagination';
+import type { PaginationItemClassKey, PaginationItemProps } from './PaginationItem';
+import type { RatingClassKey, RatingProps } from './Rating';
+import type { SkeletonClassKey, SkeletonProps } from './Skeleton';
+import type { SpeedDialClassKey, SpeedDialProps } from './SpeedDial';
+import type { SpeedDialActionClassKey, SpeedDialActionProps } from './SpeedDialAction';
+import type { SpeedDialIconClassKey, SpeedDialIconProps } from './SpeedDialIcon';
+import type { TabListClassKey, TabListProps } from './TabList';
+import type { TabPanelClassKey, TabPanelProps } from './TabPanel';
+import type { ToggleButtonClassKey, ToggleButtonProps } from './ToggleButton';
+import type { ToggleButtonGroupClassKey, ToggleButtonGroupProps } from './ToggleButtonGroup';
+import type { TreeItemClassKey, TreeItemProps } from './TreeItem';
+import type { TreeViewClassKey, TreeViewProps } from './TreeView';
+
+/**
+ * This can merged into ComponentNameToClassKey via TypeScript interface merging
+ * to provide types for overriding lab components in the theme.
+ */
+export interface LabComponentNameToClassKey {
+  MuiAlert: AlertClassKey;
+  MuiAlertTitle: AlertTitleClassKey;
+  MuiAutocomplete: AutocompleteClassKey;
+  MuiAvatarGroup: AvatarGroupClassKey;
+  MuiPagination: PaginationClassKey;
+  MuiPaginationItem: PaginationItemClassKey;
+  MuiRating: RatingClassKey;
+  MuiSkeleton: SkeletonClassKey;
+  MuiSpeedDial: SpeedDialClassKey;
+  MuiSpeedDialAction: SpeedDialActionClassKey;
+  MuiSpeedDialIcon: SpeedDialIconClassKey;
+  MuiTabList: TabListClassKey;
+  MuiTabPanel: TabPanelClassKey;
+  MuiToggleButton: ToggleButtonClassKey;
+  MuiToggleButtonGroup: ToggleButtonGroupClassKey;
+  MuiTreeItem: TreeItemClassKey;
+  MuiTreeView: TreeViewClassKey;
+}
+
+export interface LabComponentsPropsList {
+  MuiAlert: AlertProps;
+  MuiAlertTitle: AlertTitleProps;
+  // MuiAutocomplete: AutocompleteProps; // TODO Best way to handle generics here?
```

Comment:
Not sure the best way to handle these generics. We could use `never` for `T`. I can't think of a reason why any value handling would be done at the theme level. But I haven't worked at all with Autocomplete....yet. It's in my top 5 todos.
The other generics I think we could default to `true`. Since they will be partial and, I think, defaulted to false by default this should work.

Reply:
We should add some type test to see what the implications would be. Intuitively I would pass `any` to the generic arguments.